packages/plugins/other/visitor-plugin-common/src/base-resolvers-visitor.ts

```diff
@@ -1519,122 +1524,133 @@ export class BaseResolversVisitor<
     return `ParentType extends ${parentType} = ${parentType}`;
   }
 
-  FieldDefinition(node: FieldDefinitionNode, key: string | number, parent: any): FieldDefinitionPrintFn {
+  FieldDefinition(node: FieldDefinitionNode, key: string | number, parent: any): FieldDefinitionResult {
     const hasArguments = node.arguments && node.arguments.length > 0;
     const declarationKind = 'type';
 
-    return (parentName, avoidResolverOptionals) => {
-      const original: FieldDefinitionNode = parent[key];
-      const parentType = this.schema.getType(parentName);
-      const meta: ReturnType<FieldDefinitionPrintFn>['meta'] = {};
+    const original: FieldDefinitionNode = parent[key];
 
-      if (this._federation.skipField({ fieldNode: original, parentType })) {
-        return { value: null, meta };
-      }
+    return {
+      node: original,
+      printContent: (parentNode, avoidResolverOptionals) => {
+        const parentName = parentNode.name as unknown as string;
```

Comment:
I see a lot of type cast like `as any as string` across this plugin. 
Actually, I'm not sure what's causing it. 
I vaguely remember seeing _something_ changing the type (Maybe new vs old visitor?) but can't exactly remember what. 😅 
Would appreciate some pointers

Reply:
Put a FIXME here: https://github.com/dotansimha/graphql-code-generator/pull/10287/files#r2086739962